### PR TITLE
Allowing file submissions

### LINKED ISSUE #2: make more generic
We should support uploading any filetypes.  If it's an image, show the image on upload, if it's an unknown, just show the filename that was uploaded with a checkmark.

We should also rename it.

### PR DESCRIPTION
**Pull request by Belelros**
When an image is uploaded a preview will be shown whereas files will show a success message.

Fixes #2

### CODE REVIEW

File: lib/image-uploader.js [hunk 1]
```diff
@@ -6,8 +6,9 @@
       method: 'POST',
       postKey: 'image',
       progressTemplate: '<div class="progress">Uploading...</div>',
-      completeTemplate: '<img/>',
-      allow: ['jpg', 'png', 'bmp', 'gif', 'jpeg'],
+      completeTemplateImage: '<img/>',
+      completeTemplateOther: '<p>The file has been uploaded &#x2714;</p>',
+      images: ['jpg', 'png', 'bmp', 'gif', 'jpeg'],
```
Comment (jgallen23) [inline lib/image-uploader.js]: I think we still want an allow array

File: lib/image-uploader.js [hunk 2]
```diff
@@ -67,11 +59,13 @@
       this.el.on('mousemove', function(e) {
         var h = input.height();
         var w = input.width();
-        if (typeof e.pageY == 'undefined' && typeof e.clientX == 'number' && document.documentElement) {
+
+				if (typeof e.pageY == 'undefined' && typeof e.clientX == 'number' && document.documentElement) {
```
Comment (jgallen23) [inline lib/image-uploader.js]: Looks like spacing got off here

File: lib/image-uploader.js [hunk 3]
```diff
@@ -87,76 +81,71 @@
...
+		getExtension: function(file) {
+			return file.split('.').pop().toLowerCase();
+		},
+
+    showComplete: function(data) {
+			var extension = this.getExtension(data);
+      var file = (this.processData) ? this.processData(data) : data;
+			var isImage = this.images.indexOf(extension) > -1;
+
+			if (file) {
```
Comment (jgallen23) [inline lib/image-uploader.js]: More funky spacing

**Comment by jgallen23** [comment]
We need to figure out a name that isn't taken by bower

**Comment by Belelros** [comment]
`file-uploader` isn't (surprisingly) taken

**Comment by jgallen23** [comment]
Huh. That is surprising

File: lib/image-uploader.js [hunk 4]
```diff
@@ -52,8 +52,8 @@
         })
         .on('change', function(e) {
           var filename = e.target.value;
-          var ext = filename.split('.').pop().toLowerCase();
-          if ($.inArray(ext, self.allow) == -1) {
+
+          if (!this.checkType(filename)) {
             alert('Please select a photo with a ' + self.allow.join(', ') + ' extension');
```
Comment (jgallen23) [inline lib/image-uploader.js]: photo -> file

File: bower.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "image-uploader",
-  "description": "jQuery image upload plugin",
-  "version": "0.4.0",
+  "name": "file-uploader",
+  "description": "jQuery file upload plugin",
+  "version": "0.5.0",
   "homepage": "https://github.com/jgallen23/image-uploader/",
```
Comment (jgallen23) [inline bower.json]: update this

File: bower.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
   "license": "MIT",
   "copyright": "Greg Allen",
```
Comment (jgallen23) [inline bower.json]: and this

File: lib/image-uploader.js [hunk 7]
```diff
@@ -85,32 +87,37 @@
         });
     },
 
+    checkType: function(file) {
+      return $.inArray(this.getExtension(file), this.allow) !== -1 || !this.allow.length;
```
Comment (jgallen23) [inline lib/image-uploader.js]: we should check allow.length first

**Comment by jgallen23** [comment]
looks good, after those tweaks are done, can you merge pull request, do a release and add to bower?

**Comment by jgallen23** [comment]
we should update the readme too

**Comment by Belelros** [comment]
It has been released and it's also in bower

--

*Antonio Laguna Matías*

**Pull request merged**